### PR TITLE
Add endpoint for predefined catalogue item properties #233

### LINKED ISSUE #233: Add endpoint for GET list of predefined catalogue item properties

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description
- Add collection for catalogue item property templates 
- Add get endpoint for getting all catalogue item property templates

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
closes #233

### CODE REVIEW

**Comment by VKTB** [comment]
Quick question: How are these templates going to be added?

File: inventory_management_system_api/models/catalogue_item_property_template.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+"""
+
+from typing import Optional
+
+from pydantic import BaseModel, ConfigDict, Field
+
+from inventory_management_system_api.models.catalogue_category import AllowedValues
+from inventory_management_system_api.models.custom_object_id_data_types import StringObjectIdField
+
+# pylint: disable=duplicate-code
```
Comment (joelvdavies) [inline inventory_management_system_api/models/catalogue_item_property_template.py:12]: Could you enable this at the end of the file again, its easy to forget about if the file is added to.

**Comment by joshuadkitenge** [comment]
> Quick question: How are new templates going to be added?

It is going to be imported like the units. However I have time to implement a full CRUD (frontend and backend) for this before MVP. I can start the implement of this after I have done the UI changes for the migration of properties which I'm starting today.

File: inventory_management_system_api/schemas/catalogue_item_property_template.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+from typing import Optional
+
+from pydantic import BaseModel, Field
+
+
+from inventory_management_system_api.schemas.catalogue_category import AllowedValuesSchema, CatalogueItemPropertyType
+
+
+# pylint: disable=duplicate-code
```
Comment (joelvdavies) [inline inventory_management_system_api/schemas/catalogue_item_property_template.py:13]: Again could you enable this at the end of the file please

**Review by joelvdavies — COMMENTED** [review]
Two minor things. Also the readme/test data dump needs updating again. I made #236, to try and make this all simpler in the future, so I could also update it there.

File: inventory_management_system_api/models/catalogue_item_property_template.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+
+class CatalogueItemPropertyTemplateOut(BaseModel):
+    """
+    Model representing a catalogue item property template.
+    """
+
+    id: StringObjectIdField = Field(alias="_id")
```
Comment (joelvdavies) [inline inventory_management_system_api/models/catalogue_item_property_template.py:20]: I see on the front end you just inherit and add the id. Cant you do the same with the model and schema here? The schema in particular also has some validation that would be needed if doing a create endpoint. Of course Viktor's PR will be adding IDs as well so perhaps its even simpler.

**Comment by VKTB** [comment]
> It is going to be imported like the units. However I have time to implement a full CRUD (frontend and backend) for this before MVP. I can start the implement of this after I have done the UI changes for the migration of properties which I'm starting today.

If this can be delivered as it is then I would rather address the full CRUD after the MVP. I think we discussed that this and other related functionality make sense to be in a separate service since it is only of use to the frontend but not the IMS API itself.

**Comment by joshuadkitenge** [comment]
#245

**Pull request closed without merging**